In [1]:
import numpy as np
from math import factorial

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

## Task1 ##

Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

### Решение ###

Количество испытаний небольшое и вероятность благоприяного исхода достаточно велика, поэтому применяем формулу Бернулли

In [5]:
P100_85 = combinations(100, 85) * (0.8**85) * (0.2 ** (100 - 85))
P100_85

0.048061793700746556

Ответ: 4.8%

## Task2 ##

Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

### Решение ###

Количество испытаний достаточно большое, но вероятность благоприяного исхода низкая, поэтому применяем формулу Пуассона

Считаем L - среднее количество наступления события перегораниий лампочек в первый день на 5000 лампочек

In [9]:
L = 0.0004 * 5000


2.0

In [12]:
P0 = (L**0) / factorial(0) * (2.72 ** -L)
P0

0.13516435986159167

Ответ на первый вопрос: 13.5%

In [14]:
P2 = (L**2) / factorial(2) * (2.72 ** -L)
P2

0.27032871972318334

Ответ на 2-й вопрос: 27%

## Task3 ##

Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

### Решение ###

Применяем формулу Бернулли

In [17]:
P144_70 = combinations(144, 70) * (0.5**70) * (0.5**(144-70))
P144_70

0.06281178035144776

Ответ: 6.3%

## Task4 ##

В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

### Решение ###

#### Вопрос 1 ####

На первый вопрос ответим через условную вероятность. 
Вероятность выбора 2-х белых из первой корзины:

In [18]:
p1 = (7/10) * (6/9)
p1

0.4666666666666666

Вероятность выбора 2-х белых из второй корзины:

In [19]:
p2 = (9/11) * (8/10)
p2

0.6545454545454547

Вероятность одновременного выбора 2-х белых мячей из 1-й корзины и 2-х белых мячей из 2-й корзины:

In [20]:
p = p1 * p2
p

0.3054545454545455

Ответ на первый вопрос: 30.5%

#### Вопрос 2 ####

всего 3 варианта
1. в 1-й корзине выбрали 2 белых мяча и 0 белых мячей во второй корзине, верояность такого события:

In [28]:
p1 = (7/10) * (6/9) * (2/11) * (1/10)
p1

0.008484848484848484

2. в 1-й корзине выбрали 1 белый мяч и 1 белый мяч во второй корзине, верояность такого события:

In [29]:
p2 = (7/10) * (3/9) * (9/11) * (2/10)
p2

0.03818181818181818

3. в 1-й корзине выбрали 0 белых мячей и 2 белый мяча во второй корзине, верояность такого события:

In [30]:
p3 = (3/10) * (2/9) * (9/11) * (8/10)
p3

0.04363636363636364

Общая вероятность:

In [31]:
p = p1 + p2 + p3
p

0.0903030303030303

Ответ на 2-й вопрос: 9%

#### Вопрос 3 ####

Найдем вероятность того, что все мячи черные:

In [32]:
p1 = (3/10) * (2/9) * (2/11) * (1/10) 
p1

0.0012121212121212121

Вероятность того, что хотя бы один мяч белый:

In [33]:
p2 = 1 - p1
p2

0.9987878787878788

Ответ на 3-й вопрос: 99.9%